kivakit-configuration/src/main/java/com/telenav/kivakit/configuration/ComponentMixin.java

```diff
@@ -0,0 +1,74 @@
+package com.telenav.kivakit.configuration;
+
+import com.telenav.kivakit.configuration.lookup.InstanceIdentifier;
+import com.telenav.kivakit.configuration.settings.Settings;
+import com.telenav.kivakit.filesystem.Folder;
+import com.telenav.kivakit.kernel.language.mixin.Mixin;
+import com.telenav.kivakit.kernel.messaging.repeaters.RepeaterMixin;
+import com.telenav.kivakit.resource.resources.packaged.Package;
+
+/**
+ * A mixin for {@link BaseComponent} which can be used by a class that already extends another base class.
+ *
+ * @author jonathanl (shibo)
+ * @see Component
+ * @see Mixin
+ */
+public interface ComponentMixin extends Component, Mixin, RepeaterMixin
```

Comment:
I am worried usage of Mixin will increase complexity of type structure. Previously, I understand XxxMixin could mix with any other type. I am thinking since extends has Mixin, why not only Repeater.

Reply:
If ComponentMixin didn't extend RepeaterMixin, then every type that used ComponentMixin would just have to extend RepeaterMixin as well to implement Component.